extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/automatic_speech_recognition.py

```diff
@@ -52,14 +55,224 @@ async def serialize(
+
+        asr_pipeline = self.pipelines[model_name]
+        completion_data = await self.deserialize(prompt, aiconfig, parameters)
+
+        response = asr_pipeline(**completion_data)
+
+        # response is a list of text outputs. This can be tested by running an asr pipeline and noticing the outputs are a list of text.
+        outputs = construct_outputs(response)
+
+        prompt.outputs = outputs
+        await aiconfig.callback_manager.run_callbacks(CallbackEvent("on_run_complete", __name__, {"result": prompt.outputs}))
+        return prompt.outputs
+
+    def _get_device(self) -> str:
+        if torch.cuda.is_available():
+            return "cuda"
+        # Mps backend is not supported for all asr models. Seen when spinning up a default asr pipeline which uses facebook/wav2vec2-base-960h 55bb623
+        return "cpu"
+
+    def get_output_text(self, response: dict[str, Any]) -> str:
+        raise NotImplementedError("get_output_text is not implemented for HuggingFaceAutomaticSpeechRecognition")
+
+
+def validate_attachment_type_is_audio(attachment: Attachment):
+    if not hasattr(attachment, "mime_type"):
+        raise ValueError(f"Attachment has no mime type. Specify the audio mimetype in the aiconfig")
+
+    if not attachment.mime_type.startswith("audio/"):
+        raise ValueError(f"Invalid attachment mimetype {attachment.mime_type}. Expected audio mimetype.")
+
+
+def validate_and_retrieve_audio_from_attachments(prompt: Prompt) -> list[str]:
+    """
+    Retrieves the audio uri's from each attachment in the prompt input.
+
+    Throws an exception if
+    - attachment is not audio
+    - attachment data is not a uri
+    - no attachments are found
+    - operation fails for any reason
+    """
+
+    if not hasattr(prompt.input, "attachments") or len(prompt.input.attachments) == 0:
+        raise ValueError(f"No attachments found in input for prompt {prompt.name}. Please add an audio attachment to the prompt input.")
+
+    audio_uris: list[str] = []
+
+    for i, attachment in enumerate(prompt.input.attachments):
+        validate_attachment_type_is_audio(attachment)
+
+        try:
+            # load the attachment data into `AttachmentInputDataWithStringValue`
+            # TODO: @ankush-lastmile update AIConfig sdk schema to include this type (can also remove this try catch once done)
+            _ = AttachmentInputDataWithStringValue(**attachment.data)
+        except Exception as e:
+            # pylint: disable=W0707
+            raise ValueError(
+                f"Attachment #{i} for prompt {prompt.name} is not a valid attachment for Automatic Speech Recognition model parser. "
+                f"Error: {str(e)}. Please specify a kind and value for the attachment data."
+            )
+        # TODO: once previous todo gets resolved, modify this line to `.kind`` instead of .data.get("kind"). It will be a pydantic base class.
+        if not attachment.data.get("kind") == "file_uri":
```

Comment:
We'll need to update the client rendering and prompt schema format to support this. And we'll need to be consistent with this type in the image-to-text parser as well

Reply:
Can we just have this be a string and migrate to the object with kind and value afterwards? My concerns here are:
- client rendering is currently adding attachments as {data: <url>, mime_type}
- image-to-text parser is currently not doing this {kind/value} stuff and that inconsistency is going to cause a lot more work on the client

Reply:
discussed offline with @rholinshead, reverted change to input data type. For ease of use we will have data as just a string and implement structured input later